File: backoffice/src/diningTable/domain/exceptions/InvalidDiningTableIdException.py
"""
 *
 * Libraries 
 *
"""

from src.shared.domain import DomainException

"""
 *
 * Classes 
 *
"""

class InvalidDiningTableIdException( DomainException ):

    """
     *
     * Constants 
     *
    """

    __INVALID_DINING_TABLE_ID = 145

    """
     *
     * Methods 
     *
    """

    def __init__( self, id : str ) -> None:
        super().__init__(
            self.__INVALID_DINING_TABLE_ID,
            'The dining table id {} is invalid'.format( id )
        )

File: preparation/src/diningTable/domain/exceptions/InvalidDiningTableIdException.cs
/* 
 * 
 * Libraries
 *
*/

using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.src.diningTable.domain {

    public class InvalidDiningTableIdException : DomainException {

        /* 
         * 
         * Constants
         *
        */

        private const int INVALID_DINING_TABLE_ID = 145;

        /* 
         * 
         * Methods
         *
        */

        public InvalidDiningTableIdException( string id ) : base( 
            INVALID_DINING_TABLE_ID,
            string.Format( "The dining table id {0} is invalid", id )
        ) {
        }

    }

}